applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,242 +60,273 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
-        }
+
         return service.lockKeyValue(baseUri.toString(), key, label, null, null);
     }
 
     /**
-     * Unlocks ConfigurationSetting. If present, label must be explicit label value (not a wildcard).
-     * For all operations it's an optional parameter. If omitted it implies null label.
+     * Unlocks ConfigurationSetting.
+     *
      * @param key key name
      * @return ConfigurationSetting
      */
     public Mono<RestResponse<ConfigurationSetting>> unlockKeyValue(String key) {
-        return unlockKeyValue(key, null, null);
+        return unlockKeyValue(key, null);
     }
 
     /**
-     * Unlocks ConfigurationSetting. If present, label must be explicit label value (not a wildcard).
-     * For all operations it's an optional parameter. If omitted it implies null label.
-     * @param key key name
-     * @return ConfigurationSetting
+     * Unlocks a ConfigurationSetting with a matching {@param key}, optional {@param label}. If present, {@param label}
+     * must be explicit label value (not a wildcard).
+     *
+     * @param key   key name
+     * @param label Optional. If not specified, {@link ConfigurationSetting#NULL_LABEL} is used. If specified, it must
+     *              be an explicit value and cannot contain wildcard characters.
+     * @return ConfigurationSetting that was unlocked.
      */
-    public Mono<RestResponse<ConfigurationSetting>> unlockKeyValue(String key, String label, ETagFilter filter) {
+    public Mono<RestResponse<ConfigurationSetting>> unlockKeyValue(String key, String label) {
         if (key == null || key.isEmpty()) {
             throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
+        } else if (label == null) {
+            label = ConfigurationSetting.NULL_LABEL;
         }
-        if (filter != null) {
-            return service.unlockKeyValue(baseUri.toString(), key, label, filter.ifMatch(), filter.ifNoneMatch());
-        }
+
         return service.unlockKeyValue(baseUri.toString(), key, label, null, null);
     }
 
     /**
-     * Lists chronological/historical representation of ConfigurationSetting resource(s). Revisions eventually expire (default 30 days).
-     * For all operations key is optional parameter. If ommited it implies any key.
-     * For all operations label is optional parameter. If ommited it implies any label.
+     * Lists the ConfigurationSettings.
+     *
      * @param filter query options
-     * @return Revisions of the ConfigurationSetting
+     * @return KeyValues
      */
-    public Flux<ConfigurationSetting> listKeyValueRevisions(RevisionFilter filter) {
-        return listKeyValueRevisions(filter, pageResponseFlux -> pageResponseFlux.map(r -> r.items())
-                                                              .flatMapIterable(i -> i));
+    public Flux<ConfigurationSetting> listKeyValues(KeyValueListFilter filter) {
+        Mono<RestResponse<PageImpl<ConfigurationSetting>>> result;
```

Comment:
We should sync up about this - is PageImpl necessary? Is this a common concept we should abstract?

Reply:
I had tried using [Page interface](https://github.com/Azure/autorest-clientruntime-for-java/blob/a55f87f3cc3a68742a2ac94c031c6d715965a9c2/azure-client-runtime/src/main/java/com/microsoft/azure/Page.java), but the deserialisation process requires a concrete class. :/ This is a common concept. @anuchandy How do the management libraries deal with this? Do they all implement the Page.java interface?